src/redux/csState.ts

```diff
@@ -49,6 +50,25 @@ export interface CsState {
   subscriptions: Subscriptions;
 }
 
+function updateValueCache(
+  oldValueCache: ValueCache,
+  newValueCache: ValueCache,
+  action: any
```

Comment:
As there are only a number of actions which will be useful for this function, perhaps the type could be specified further? I suppose it won't matter much as it will mainly be used at runtime but it might make the code easier to reason about.

Reply:
Yes, this was a lazy `any`.